op-challenger/game/fault/trace/cannon/provider.go

```diff
@@ -44,10 +48,12 @@ type ProofGenerator interface {
 }
 
 type CannonTraceProvider struct {
-	logger    log.Logger
-	dir       string
-	prestate  string
-	generator ProofGenerator
+	logger         log.Logger
+	dir            string
+	prestate       string
+	generator      ProofGenerator
+	lastStepWriter LastStepWriter
+	lastStepReader LastStepReader
```

Comment:
We don't need these to be stored as function references anymore - the production code can just directly call the methods since the tests don't need to stub them anymore.